### PR TITLE
Draft: [DOP-20059] add run_id and job_id to inputs and outputs

### PR DESCRIPTION
**Pull request by TiGrib**
## Change Summary

Add run_id and job_id to inpputs and outputs.
Update models and consumer.

## Related issue number

[DOP-20059]

## Checklist

* [x] Commit message and PR title is comprehensive
* [x] Keep the change as small as possible
* [x] Unit and integration tests for the changes exist
* [x] Tests pass on CI and coverage does not decrease
* [ ] Documentation reflects the changes where applicable
* [ ] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/data-rentgen/blob/develop/CONTRIBUTING.rst) for details.)
* [x] My PR is ready to review.

### CODE REVIEW

File: data_rentgen/server/schemas/v1/lineage.py [hunk 1]
```diff
@@ -80,18 +81,38 @@ def _check_until(cls, value: datetime | None, info: ValidationInfo) -> datetime
 
 class DatasetLineageQueryV1(BaseLineageQueryV1):
     start_node_id: int = Field(description="Dataset id", examples=[42])
+    granularity: Literal["JOB", "RUN", "OPERATION"] = Field(
```
Comment (dolfinus) [inline data_rentgen/server/schemas/v1/lineage.py:84]: Let's move granularity to a dedicated PR

File: tests/test_server/fixtures/factories/input.py [hunk 2]
```diff
@@ -11,6 +11,8 @@ def input_factory(**kwargs) -> Input:
         "id": input_id,
         "created_at": extract_timestamp_from_uuid(input_id),
         "operation_id": generate_new_uuid(),
+        "run_id": generate_new_uuid(),
```
Comment (dolfinus) [inline tests/test_server/fixtures/factories/input.py:14]: Update lineage factories to fill up these attributes instead of using random ones

File: data_rentgen/consumer/handlers.py [hunk 3]
```diff
@@ -210,20 +210,29 @@ async def get_or_create_schema(schema: SchemaDTO, unit_of_work: UnitOfWork) -> S
...
     output: OutputDTO,
     operation: Operation,
+    run: Run,
     dataset: Dataset,
     schema: Schema | None,
     unit_of_work: UnitOfWork,
 ) -> Output:
     schema_id = schema.id if schema else None
-    return await unit_of_work.output.create_or_update(output, operation.id, dataset.id, schema_id)
+    return await unit_of_work.output.create_or_update(output, operation.id, run.id, run.job_id, dataset.id, schema_id)
```
Comment (dolfinus) [inline data_rentgen/consumer/handlers.py:238]: Please add asserts to consumer tests, to check that new attributes are populated properly

Comment (TiGrib) [reply data_rentgen/consumer/handlers.py]: Update `test_runs_handler_spark.py`

File: tests/test_server/fixtures/factories/lineage.py [hunk 4]
```diff
@@ -182,6 +191,8 @@ async def lineage_with_depth(
             input_factory(
                 created_at=operation.created_at,
                 operation_id=operation.id,
+                run_id=operation.run_id,
+                job_id=next(run.job_id for run in actual_runs if run.id == operation.run_id),
```
Comment (dolfinus) [inline tests/test_server/fixtures/factories/lineage.py:195]: Just create a dict[run_id, job_id], instead of introducing inner cycles

**Review by dolfinus — APPROVED** [review]
